cdap-common/src/main/java/co/cask/cdap/common/io/Processor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.io;
+
+/**
+ * Process input values.
+ *
+ * @param <T> type of the input values
+ */
+public interface Processor<T> {
+
+  /**
+   * Process an input value.
+   *
+   * @param input input value to process
+   */
+  void process(T input);
```

Comment:
Also, you might want to consider having it look like the `LineProcessor` in guava which has a method to return result. With that, you can have the `processLocations` method looks like:

```java
interface Processor<T, R> {
  boolean process(T t);
  R getResult();
}

<R> R processLocations(Location location, Processor<LocationStatus, R> processor);
```